File: DSA_ALGO_GRIND75/ARRAY/121.cpp
#include<algorithm>
#include<vector>
#include<iostream>
using namespace std;

class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int buy = prices[0];
        int profit = 0;
        int maxx = 0;

        for(int i =1; i < prices.size(); i++){
            if (prices[i] < buy){
                buy = prices[i];
            }
            else {
                profit = prices[i] - buy;
                maxx = max(profit,maxx);
            }
        }
    }
};

File: Leetcode_Solutions/ARRAY/121.py
class Solution(object):
    def maxProfit(self, prices):
        profit = 0
        buy = prices[0]
        for sell in prices[1:]:
            if sell > buy:
                profit = max(profit, sell - buy)
            else:
                buy = sell
        
        return profit
    
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        ans = 0
        buy = prices[0]

        for price in prices:
            if price <= buy:
                buy = price

            else:
                ans = max(ans, price-buy)

        return ans